src/crud.js

```diff
@@ -0,0 +1,185 @@
+};
+
+const saveTodos = (e) => {
+  const saveBtn = e.target;
+  const existingTodos = JSON.parse(localStorage.getItem('todos'));
+  const btnClass = saveBtn.className;
+  const btnId = btnClass.split('-');
+  const id = parseInt(btnId[1], 10);
+  const taskList = document.querySelector(`#tasks-${id}`);
+  const saveEdit = document.querySelector(`.save-${id}`);
+  const deleteEdit = document.querySelector(`.delete-${id}`);
+  const inputId = document.querySelector(`#complete-${id}`);
+  const editBtn = document.getElementById(`${id}`);
+
+  editBtn.style.display = 'block';
+  inputId.style.outline = 'none';
+
+  existingTodos[id].description = inputId.value;
+  localStorage.setItem('todos', JSON.stringify(existingTodos));
+  saveEdit.remove();
+  deleteEdit.remove();
+  taskList.classList.remove('active');
+  inputId.setAttribute('readonly', true);
+};
+
+const completedTodos = (e) => {
+  const checkbox = e.target;
+  let btnId = checkbox.id;
+  btnId = btnId.split('-');
+  const id = parseInt(btnId[1], 10);
+  const inputId = document.querySelector(`#complete-${id}`);
+  const existingTodos = JSON.parse(localStorage.getItem('todos'));
+  if (checkbox.checked) {
+    existingTodos[id].completed = true;
+    inputId.style.textDecoration = 'line-through';
+    localStorage.setItem('todos', JSON.stringify(existingTodos));
+  } else {
+    existingTodos[id].completed = false;
+    inputId.style.textDecoration = 'none';
+    localStorage.setItem('todos', JSON.stringify(existingTodos));
+  }
+};
+
+const editTodos = (e) => {
+  const editInput = e.target;
+  const editBtn = document.getElementById(`${editInput.id}`);
+  const inputId = document.querySelector(`#complete-${editInput.id}`);
+  inputId.removeAttribute('readonly');
+  inputId.focus();
+
+  editBtn.style.display = 'none';
+  inputId.style.outline = 'auto';
+  inputId.style.outlineColor = 'gray';
+
+  const taskList = document.querySelector(`#tasks-${editInput.id}`);
+  taskList.classList.add('active');
+
+  const task = document.querySelector(`#task-${editInput.id}`);
+
+  const save = document.createElement('img');
+  save.setAttribute('src', saveImg);
+  save.classList.add(`save-${editInput.id}`);
+  task.appendChild(save);
+
+  const removeBtn = document.createElement('img');
+  removeBtn.classList.add(`delete-${editInput.id}`);
+  removeBtn.setAttribute('src', removeIcon);
+  task.appendChild(removeBtn);
+
+  document.querySelectorAll(`.delete-${editInput.id}`).forEach((e) => {
+    e.addEventListener('click', deleteTodos);
+  });
+
+  document.querySelectorAll(`.save-${editInput.id}`).forEach((e) => {
+    e.addEventListener('click', saveTodos);
+  });
+};
+
+const createTodos = () => {
+  const existingTodos = JSON.parse(localStorage.getItem('todos'));
+  if (existingTodos !== null && existingTodos.length > 0) {
+    const todoContainer = document.querySelector('.todo-container');
+    todoContainer.innerHTML = '';
+
+    existingTodos.forEach((todo, index) => {
```

Comment:
You can create the inner html using Domlist under the "div".

Reply:
Noted. Thank you for the feedback